plugins/cost-insights/src/types/Alert.ts

```diff
@@ -13,23 +13,147 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
+import { ForwardRefExoticComponent, RefAttributes } from 'react';
 import { ChangeStatistic } from './ChangeStatistic';
+import { Duration } from './Duration';
 import { Maybe } from './Maybe';
 
 /**
  * Generic alert type with required fields for display. The `element` field will be rendered in
  * the Cost Insights "Action Items" section. This should use data fetched in the CostInsightsApi
  * implementation to render an InfoCard or other visualization.
+ *
+ * The alert type exposes hooks which can be used to enable and access various events,
+ * such as when a user dismisses or snoozes an alert. Default forms and buttons
+ * will be rendered if a hook is defined.
+ *
+ * Each default form can be overridden with a custom component. It must be implemented using
+ * React.forwardRef. See https://reactjs.org/docs/forwarding-refs
+ *
+ * Errors thrown within hooks will generate a snackbar error notification.
  */
+
 export type Alert = {
   title: string;
   subtitle: string;
+  element?: JSX.Element;
+  status?: AlertStatus;
   url?: string;
   buttonText?: string; // Default: View Instructions
-  element: JSX.Element;
+  SnoozeForm?: Maybe<AlertForm>;
+  AcceptForm?: Maybe<AlertForm>;
+  DismissForm?: Maybe<AlertForm>;
+  onSnoozed?(options: AlertOptions): Promise<Alert[]>;
+  onAccepted?(options: AlertOptions): Promise<Alert[]>;
+  onDismissed?(options: AlertOptions): Promise<Alert[]>;
```

Comment:
Alert.element is now optional. The client can return an alert in two states - with a status or without a status. Alerts with a status property do not render action items, so its unnecessary to require this property to be present. They are however, hidden in the collapsed view - so the title and subtitle are still required.